test/libyul/yulOptimizerTests/fullSuite/unusedFunctionParameterPruner_loop.yul

```diff
@@ -25,13 +27,12 @@
 //     function f()
 //     {
 //         let b := 10
-//         let _1 := 0
-//         let a := calldataload(_1)
-//         let _2 := iszero(a)
+//         let a := calldataload(0)
+//         let _1 := iszero(a)
 //         for { } iszero(b) { b := add(b, not(0)) }
 //         {
-//             mstore(a, _1)
-//             if _2 { leave }
+//             mstore(a, 0)
```

Comment:
Actually, these tests indicate that this change already does the trick in general and that the CSE in the cleanup sequence does not destroy this again - probably due to the fact that we don't run the expression splitter again beforehand. So maybe adding ``mu`` to the cleanup sequence is even unnecessary after this (we should still check and evaluate, though).